admin/RequestSolrCloudBackups

```diff
@@ -0,0 +1,307 @@
+            grep { $backup_requests{$_}{status} eq 'running' }
+            keys %backup_requests
+        )
+        {
+            my $solr_response = _query_collections_api('REQUESTSTATUS', {
+                    requestid => $request_id,
+                }
+            );
+            _debug (
+                "Received: %s",
+                encode_json($solr_response),
+            );
+            if (exists $solr_response->{error}) {
+                _error(
+                    "Failed on backup request '%s' of collection '%s' as '%s': " .
+                    "Error '%s'",
+                    $request_id,
+                    $backup_requests{$request_id}{collection},
+                    $backup_requests{$request_id}{name},
+                    $solr_response->{error},
+                );
+                _flush_running_requests_status();
+                exit 70; # EX_SOFTWARE
+            }
+            elsif ($solr_response->{status}->{state} eq 'completed')
+            {
+                $backup_requests{$request_id}{status} = 'completed';
+                _query_collections_api('DELETESTATUS', {
+                        requestid => $request_id,
+                    }
+                );
+            }
+            elsif ($solr_response->{status}->{state} ne 'running')
+            {
+                _error(
+                    "Failed on backup request '%s' of collection '%s' as '%s': " .
+                    "Unhandled state '%s'",
+                    $request_id,
+                    $backup_requests{$request_id}{collection},
+                    $backup_requests{$request_id}{name},
+                    $solr_response->{status}->{state},
+                );
+                _flush_running_requests_status();
+                exit 70; # EX_SOFTWARE
+            }
+        }
+    }
+}
+
+sub _flush_running_requests_status
+{
+    foreach my $request_id (
+        sort
+        grep { $backup_requests{$_}{status} eq 'running' }
+        keys %backup_requests
+    )
+    {
+        _query_collections_api('DELETESTATUS', {
+                requestid => $request_id,
+            }
+        );
+    }
+}
+
+sub _query_collections_api
+{
+    my ($action, $parameters) = @_;
+
+    my $url = $COLLECTIONS_API . '?action=' . $action;
+    foreach my $key (keys %{$parameters})
+    {
+        $url .= '&' . $key . '=' . $parameters->{$key};
+    }
+    _debug "Querying '$url'";
+
+    my $http_response = $c->lwp->get($url);
+    if (!$http_response->is_success)
+    {
+        _error(
+            "Failed HTTP request '%s': Error '%s'",
+            $url,
+            $http_response->status_line,
+        );
+        exit 70; # EX_SOFTWARE
+    }
+
+    my $http_content = $http_response->decoded_content;
+    my $solr_response = decode_json($http_content);
+    if (!defined $solr_response->{responseHeader}
+        || !defined $solr_response->{responseHeader}->{status}
+        || $solr_response->{responseHeader}->{status} ne '0')
```

Comment:
Perl will autovivify the keys as needed, so you can leave out the definedness checks for readability if desired
```suggestion
    if (($solr_response->{responseHeader}{status} // '') ne '0')
```

Reply:
Applied in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/b7187807e3552f60c30c97f01310c79d31a44997